### ISSUE TITLE
[Enhancement]: Allow setting up decimals for the cart quantity inputs step

### ISSUE THREAD

**Issue by smachi** [woocommerce/woocommerce#44421]
### Describe the solution you'd like

We have a plugin ([https://wordpress.org/plugins/atum-stock-manager-for-woocommerce/](<https://wordpress.org/plugins/atum-stock-manager-for-woocommerce/>)) that adds a feature to WooCommerce to be able to use decimal numbers for the stock quantities.<br>It works flawlessly everywhere except for the new Cart and Checkout blocks.

I've tried to set this step attribute via this Store API filter "woocommerce_store_api_product_quantity_multiple_of" (by returning 0.1, for example) but it shows an error in the front end:

*Unexpected error in: woocommerce/cart-line-items-block**<br>**Error: You must provide an integer.*

Is there a way to bypass this validation?

### Describe alternatives you've considered

*No response*

### Additional context

*No response*

**Comment by nielslange** [comment]
Thanks for reaching out, @smachi. To better understand this issue, can you provide your use case? If I understand you correctly, this issue only affects the cart quantity input when using the Cart block. Is that correct? Could you provide steps to reproduce this issue, that also show how it's working in other areas? This would help us to understand the importance of this request. Thanks in advance.

**Comment by nielslange** [comment]
Related earlier discussion: https://github.com/woocommerce/woocommerce-blocks/discussions/11581

**Comment by globalhealthcare** [comment]
Hi @nielslange 

Yes, that discussion is related to this.
We had this functionality to be able to use decimals for the stock quantity in our plugin for years. And there are many of our users using it.
When you enable this feature from our plugin's settings, you can use decimals (like 1.15) on the stock quantity field within the product page and the customers can also order WooCommerce products by using decimals (but only when the legacy shortcodes are used for now).

I was not able to bypass the integer restriction when using the cart and checkout blocks though.

**Comment by smachi** [comment]
Hi @nielslange 

Yes, that discussion is related to this.
We have had this functionality to use decimals for the stock quantity in our plugin for years. And there are many of our users using it.
When you enable this feature from our plugin's settings, you can use decimals (like 1.15) on the stock quantity field within the product page and the customers can also order WooCommerce products by using decimals (but only when the legacy shortcodes are used for now).

I could not bypass the integer restriction when using the cart and checkout blocks though.

**Comment by wavvves** [comment]
Hi @smachi thanks for reporting this. As I see it this is a feature present on shortcode checkout flow blocked by Store API endpoint specs, and it would allow items to be sold by weight, volume, etc.
I'm adding this to our backlog 👍

**Comment by smachi** [comment]
Perfect. Thank you!

**Comment by pierorocca** [comment]
Hi @smachi - thanks for the report.

I know this has been open for a long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've taken a look and marked this ready for the engineering team to pick up.

What I'm seeing: Backend store API quantity schemas already handle numeric multiple_of values, but some client checkout or cart code still assumes integer-style quantity handling, so decimal step support appears incomplete on trunk.

**Comment by ralucaStan** [comment]
This was fixed in 10.1.0 via woocommerce/woocommerce#58693
